### PR TITLE
Add omniData operator offload feature.

### PR DESCRIPTION
**Pull request by jiaotongZou**
Add omniData operator offload feature.

### CODE REVIEW

File: connector/src/main/java/io/prestosql/plugin/hive/HiveConfig.java [hunk 1]
```diff
@@ -1892,4 +1901,113 @@ public boolean getWorkerMetaStoreCacheEnabled()
...
     }
+
+    public String getOmniDataServerTarget()
+    {
+        return this.omniDataServerTarget;
+    }
+
+    @Config("omni-data.server.target")
+    @ConfigDescription("the ip and port of server target")
+    public HiveConfig setOmniDataServerTarget(String omniDataServerTarget)
```
Comment (luodan000) [inline connector/src/main/java/io/prestosql/plugin/hive/HiveConfig.java:1912]: 建议把omniDataServerTarget删除点，target地址是从split的address获取的，这个是之前demo的时候的使用方法

File: connector/src/main/java/io/prestosql/plugin/hive/HiveConfig.java [hunk 2]
```diff
@@ -1892,4 +1901,113 @@ public boolean getWorkerMetaStoreCacheEnabled()
...
+    @Config("hive.ipu-https-required")
+    @ConfigDescription("Enables https communication.")
+    public HiveConfig setHttpsRequired(boolean httpsRequired)
+    {
+        this.httpsRequired = httpsRequired;
+        return this;
+    }
+
+    public boolean isHttpsRequired()
+    {
```
Comment (luodan000) [inline connector/src/main/java/io/prestosql/plugin/hive/HiveConfig.java:2010]: 请确认这个配置是否被使用到？

File: connector/src/main/java/io/prestosql/plugin/hive/HivePushDownRecordPageSource.java [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    {
+        if (closed) {
+            return;
+        }
+        closed = true;
+        try {
+            dataReader.close();
+        }
+        catch (Exception e) {
+            e.printStackTrace();
```
Comment (luodan000) [inline connector/src/main/java/io/prestosql/plugin/hive/HivePushDownRecordPageSource.java:102]: 请确认是否打印stack trace

File: connector/src/main/java/io/prestosql/plugin/hive/ipu/IpuNodeManager.java [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+
+    private final ScheduledExecutorService nodeStateUpdateExecutor;
+
+    private KerberosConfig kerberosConfig;
+    private HttpClientConfig httpClientConfig;
+    private DiscoveryClientConfig discoveryClientConfig;
+    private final boolean httpsRequired;
+
+    @Inject
+    public IpuNodeManager()
```
Comment (luodan000) [inline connector/src/main/java/io/prestosql/plugin/hive/ipu/IpuNodeManager.java:80]: 记一个遗留问题，Ipu这个词需要rename

File: connector/src/main/java/io/prestosql/plugin/hive/ipu/IpuNodeManager.java [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+    {
+        try (HttpClient httpClient = new JettyHttpClient("ipu-node-manager", httpClientConfig, kerberosConfig, ImmutableList.of())) {
+            URI uri = discoveryClientConfig.getDiscoveryServiceURI();
+            if (uri == null) {
+                throw new DiscoveryException("No discovery servers are available");
+            }
+
+            String type = "ipu";
+            uri = URI.create(uri + "/v1/service/" + type + "/");
+
```
Comment (luodan000) [inline connector/src/main/java/io/prestosql/plugin/hive/ipu/IpuNodeManager.java:158]: 当前往discovery注册的类型都是type = “ipu”，需要和omnidata server侧同时修改为“omnidata”或者其他

File: src/main/java/io/prestosql/plugin/hive/HivePageSourceProvider.java [hunk 6]
```diff
@@ -102,6 +122,7 @@ public HivePageSourceProvider(
             Set<HiveSelectivePageSourceFactory> selectivePageSourceFactories)
     {
         requireNonNull(hiveConfig, "hiveConfig is null");
+        this.ipuNodeManager = ipuNodeManager;
```
[the remark this answers is not in the document]
Comment (jiaotongZou) [reply src/main/java/io/prestosql/plugin/hive/HivePageSourceProvider.java]: done

[the remark this answers is not in the document]
Comment (jiaotongZou) [reply src/main/java/io/prestosql/plugin/hive/HivePageSourceProvider.java]: done

File: src/main/java/io/prestosql/plugin/hive/HivePushDownRecordPageSource.java [hunk 7]
```diff
@@ -0,0 +1,105 @@
+/*
```
[the remark this answers is not in the document]
Comment (jiaotongZou) [reply src/main/java/io/prestosql/plugin/hive/HivePushDownRecordPageSource.java]: done

File: connector/src/main/java/io/prestosql/plugin/hive/HiveConfig.java (hunk 1 above)
Comment (jiaotongZou) [reply connector/src/main/java/io/prestosql/plugin/hive/HiveConfig.java]: deleted.

File: connector/src/main/java/io/prestosql/plugin/hive/HiveConfig.java (hunk 2 above)
Comment (jiaotongZou) [reply connector/src/main/java/io/prestosql/plugin/hive/HiveConfig.java]: I have deleted it.

File: connector/src/main/java/io/prestosql/plugin/hive/HivePushDownRecordPageSource.java (hunk 3 above)
Comment (jiaotongZou) [reply connector/src/main/java/io/prestosql/plugin/hive/HivePushDownRecordPageSource.java:106]: done. We will handle the exception as ORCPageSource.

File: connector/src/main/java/io/prestosql/plugin/hive/ipu/IpuNodeManager.java (hunk 5 above)
Comment (jiaotongZou) [reply connector/src/main/java/io/prestosql/plugin/hive/ipu/IpuNodeManager.java]: I've renamed all the 'ipu' strings.

File: src/main/java/io/prestosql/plugin/hive/ipu/CommunicationConfig.java [hunk 8]
```diff
@@ -0,0 +1,33 @@
+/*
```
[the remark this answers is not in the document]
Comment (jiaotongZou) [reply src/main/java/io/prestosql/plugin/hive/ipu/CommunicationConfig.java]: done

File: connector/src/main/java/io/prestosql/plugin/hive/ipu/IpuNodeManager.java (hunk 4 above)
Comment (jiaotongZou) [reply connector/src/main/java/io/prestosql/plugin/hive/ipu/IpuNodeManager.java]: done

**Pull request merged**